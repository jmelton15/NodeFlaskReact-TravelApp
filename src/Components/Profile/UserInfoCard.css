.UserInfoCard-Card {
    background-color: #E6E6FA !important;
    box-shadow: 5px 30px 15px;
}

#userinfocard-editBioIcon:hover {
    cursor: pointer;
}

#userinfocard-username{
    font-family: "AestheticViolet";
    font-weight: bold;
}

#userinfocard-followingInfo {
    display: flex;
    justify-content: center;
}

#userinfocard-followers{
    margin-right: 30px;
}
#userinfocard-followers:hover{
    cursor: pointer;
}
#userinfocard-followers span {
    text-decoration: underline;
    font-weight: bold;
}

#userinfocard-following:hover {
    cursor: pointer;
}
#userinfocard-following span {
    text-decoration: underline;
    font-weight: bold;
}

#userinfocard-followBtn {
    border-radius: 30px !important;
    background-color: #dcead0;
    color: black;
    font-weight: bold;
}

#userinfocard-followBtn:hover {
    background-color: #a8c78d;
}


#alreadyFollowing {
    color:#21ADA8;
    margin-right: 10px;
    font-weight: bold;
}

#userinfocard-journalIcon:hover {
 cursor: pointer;
}


@media only screen and (min-device-width: 5px) and (max-device-width: 420px) {
    #userinfocard-profilePic {
        height:250px;
    }
}


