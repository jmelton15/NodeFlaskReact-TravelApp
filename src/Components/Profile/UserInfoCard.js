import "./UserInfoCard.css";
import FileUploadForm from "./FileUploadForm";
import {
    Card, CardImg, CardText, CardBody,
    CardTitle, CardSubtitle, Tooltip
  } from 'reactstrap';
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faBook,faEdit} from "@fortawesome/free-solid-svg-icons";
import { useState } from "react";
import BioUpdateForm from "./BioUpdateForm";
import {NodeApi} from "../../APIRequests/nodeApi";

const UserInfoCard = ({user,goToPage,profilePic,uploadPicture,getConnections,token}) => {
    const [bio,setBio] = useState(user.bio);

    const [journalTTOpen, setJournalTTOpen] = useState(false);
    const toggleJournalToolTip = () => setJournalTTOpen(!journalTTOpen);

    const [bioTTOpen, setBioTTOpen] = useState(false);
    const toggleBioToolTip = () => setBioTTOpen(!bioTTOpen);

    const [showEditBio,setShowEditBio] = useState(false);
    const toggleBioForm = () => setShowEditBio(!showEditBio);


    const updateBio = async (bioTxt) => {
        let newBio = await NodeApi.editBio(user.user_id,bioTxt.bioTxt,token);
        setBio(newBio.bio)
        toggleBioForm();
        getConnections();
    }
    
    return (
        <div className="UserInfoCard-CardContainer">
            <Card className="UserInfoCard-Card">
                <CardImg top width="100%" src={profilePic} alt="Profile" />
                <CardBody>
                    <div id="userinfocard-usernameBio">
                        <div className="d-flex justify-content-center">
                            <CardTitle className="text-center me-3" tag="h1" id="userinfocard-username">{user.username.toUpperCase()}</CardTitle>
                            <FontAwesomeIcon icon={faBook} 
                                            size="2x" 
                                            color="#E97451" 
                                            id="userinfocard-journalIcon" 
                                            onClick={() => goToPage(`users/${user.user_id}/traveljournal`)}
                            />
                            <Tooltip toggle={toggleJournalToolTip} isOpen={journalTTOpen} target="userinfocard-journalIcon">Open Travel Journal!</Tooltip>
                        </div>
                        <div className="d-flex justify-content-between align-items-center">
                            {showEditBio && <BioUpdateForm updateBio={updateBio}/>}
                            {!showEditBio && <CardSubtitle tag="h6" className="mb-2 text-muted w-100">{bio}</CardSubtitle>}
                            <FontAwesomeIcon id="userinfocard-editBioIcon" icon={faEdit} onClick={toggleBioForm}/>
                            <Tooltip toggle={toggleBioToolTip} isOpen={bioTTOpen} target="userinfocard-editBioIcon">Edit Bio</Tooltip>
                        </div>
                    </div>
                    <CardText id="userinfocard-followingInfo">
                        <p id="userinfocard-followers" onClick={() => goToPage(`users/${user.user_id}/followers`)}><span>Followers:</span> {user.follower_count}</p>
                        <p id="userinfocard-following" onClick={() => goToPage(`users/${user.user_id}/following`)}><span>Following:</span> {user.follow_count}</p>
                    </CardText>
                    <FileUploadForm uploadPicture={uploadPicture}/>
                </CardBody>
            </Card>
        </div>
    )
}

export default UserInfoCard;