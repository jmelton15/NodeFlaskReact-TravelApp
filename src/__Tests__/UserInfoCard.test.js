import {render } from '@testing-library/react';
import {user} from "./TestHelperObjects";
import UserInfoCard from "../Components/Profile/UserInfoCard";


// simple smoke test for UserInfoCard component
it("renders a UserInfoCard component without crashing", function() {
    render(<UserInfoCard user={user} token={true} goToPage={() => ""} profilePic={""} uploadPicture={() => ""} getConnections={() => ""}/>);
});

// simple snapshot test for UserInfoCard Component
it("should match snapshot for UserInfoCard", function() {
    const {asFragment} = render (<UserInfoCard user={user} token={true} goToPage={() => ""} profilePic={""} uploadPicture={() => ""} getConnections={() => ""}/>);
    expect(asFragment()).toMatchSnapshot();
});
