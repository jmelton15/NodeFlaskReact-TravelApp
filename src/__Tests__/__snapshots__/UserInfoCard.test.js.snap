// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot for UserInfoCard 1`] = `
<DocumentFragment>
  <div
    class="UserInfoCard-CardContainer"
  >
    <div
      class="UserInfoCard-Card card"
    >
      <img
        alt="Profile"
        class="card-img-top"
        id="userinfocard-profilePic"
        src=""
        width="100%"
      />
      <div
        class="card-body"
      >
        <div
          id="userinfocard-usernameBio"
        >
          <div
            class="d-flex justify-content-center"
          >
            <h1
              class="text-center me-3 card-title"
              id="userinfocard-username"
            >
              TEST1
            </h1>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-book fa-w-14 fa-2x "
              color="#E97451"
              data-icon="book"
              data-prefix="fas"
              focusable="false"
              id="userinfocard-journalIcon"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z"
                fill="currentColor"
              />
            </svg>
          </div>
          <div
            class="d-flex justify-content-between align-items-center"
          >
            <h6
              class="mb-2 text-muted w-100 card-subtitle"
            >
              Hi! Im Test1
            </h6>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-edit fa-w-18 "
              data-icon="edit"
              data-prefix="fas"
              focusable="false"
              id="userinfocard-editBioIcon"
              role="img"
              viewBox="0 0 576 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
        <p
          class="card-text"
          id="userinfocard-followingInfo"
        />
        <p
          id="userinfocard-followers"
        >
          <span>
            Followers:
          </span>
           0
        </p>
        <p
          id="userinfocard-following"
        >
          <span>
            Following:
          </span>
           0
        </p>
        <p />
        <form
          class="mt-3"
        >
          <div
            class="d-flex flex-column form-group"
          >
            <label
              class=""
              for="fileInput"
            >
              Change Profile Picture
            </label>
            <input
              class="form-control-file"
              id="fileInput"
              name="fileInput"
              type="file"
              value=""
            />
            <button
              class="mt-2 btn-sm btn btn-secondary"
            >
              Submit Picture Upload
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
